source/extensions/filters/common/lua/protobuf_converter.cc

```diff
@@ -0,0 +1,214 @@
+      ProtobufConverterUtils::messageToLuaTable(
+          state_, reflection->GetRepeatedMessage(*message_, field_, index));
+      break;
+    case Protobuf::FieldDescriptor::CPPTYPE_INT32:
+    case Protobuf::FieldDescriptor::CPPTYPE_INT64:
+      lua_pushnumber(state_, reflection->GetRepeatedInt64(*message_, field_, index));
+      break;
+    case Protobuf::FieldDescriptor::CPPTYPE_BOOL:
+      lua_pushboolean(state_, reflection->GetRepeatedBool(*message_, field_, index));
+      break;
+    default:
+      lua_pushnil(state_);
+      break;
+    }
+  }
+
+  lua_State* state_;
+  const Protobuf::ReflectableMessage& message_;
+  const Protobuf::FieldDescriptor* field_;
+};
+
+} // namespace
+
+void ProtobufConverterUtils::messageToLuaTable(lua_State* state, const Protobuf::Message& message) {
+  lua_newtable(state);
+
+  Protobuf::ReflectableMessage reflectable_message = createReflectableMessage(message);
+  const auto* reflection = reflectable_message->GetReflection();
+
+  std::vector<const Protobuf::FieldDescriptor*> fields;
+  reflection->ListFields(*reflectable_message, &fields);
+
+  for (const auto* field : fields) {
+    pushLuaString(state, field->name());
+
+    if (field->is_map()) {
+      MapFieldConverter(state, reflectable_message, field).convert();
+    } else if (field->is_repeated()) {
+      RepeatedFieldConverter(state, reflectable_message, field).convert();
+    } else {
+      fieldToLuaValue(state, message, field);
+    }
+
+    lua_settable(state, -3);
+  }
+}
+
+void ProtobufConverterUtils::fieldToLuaValue(lua_State* state, const Protobuf::Message& message,
+                                             const Protobuf::FieldDescriptor* field) {
+  Protobuf::ReflectableMessage reflectable_message = createReflectableMessage(message);
+  const auto* reflection = reflectable_message->GetReflection();
+
+  switch (field->cpp_type()) {
+  case Protobuf::FieldDescriptor::CPPTYPE_INT32:
+  case Protobuf::FieldDescriptor::CPPTYPE_INT64:
+    lua_pushnumber(state, reflection->GetInt64(*reflectable_message, field));
+    break;
+  case Protobuf::FieldDescriptor::CPPTYPE_UINT32:
+  case Protobuf::FieldDescriptor::CPPTYPE_UINT64:
+    lua_pushnumber(state, reflection->GetUInt64(*reflectable_message, field));
+    break;
+  case Protobuf::FieldDescriptor::CPPTYPE_DOUBLE:
+    lua_pushnumber(state, reflection->GetDouble(*reflectable_message, field));
+    break;
+  case Protobuf::FieldDescriptor::CPPTYPE_FLOAT:
+    lua_pushnumber(state, reflection->GetFloat(*reflectable_message, field));
+    break;
+  case Protobuf::FieldDescriptor::CPPTYPE_BOOL:
+    lua_pushboolean(state, reflection->GetBool(*reflectable_message, field));
+    break;
+  case Protobuf::FieldDescriptor::CPPTYPE_STRING:
+    pushLuaString(state, reflection->GetString(*reflectable_message, field));
+    break;
+  case Protobuf::FieldDescriptor::CPPTYPE_MESSAGE:
+    messageToLuaTable(state, reflection->GetMessage(*reflectable_message, field));
```

Comment:
To mirror the lua api naming, this might be clearer if you rename the function to `pushLuaTableFromMessage` or some other variation starting with `push`. Same for other similar methods.